drivers/timer/arcv2_timer0.c

```diff
@@ -512,76 +178,69 @@ int z_clock_driver_init(struct device *device)
 	return 0;
 }
 
-#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
-static int sys_clock_suspend(struct device *dev)
+void z_clock_set_timeout(s32_t ticks, bool idle)
 {
-	ARG_UNUSED(dev);
+	/* If the kernel allows us to miss tick announcements in idle,
+	 * then shut off
+	 * the counter. (Note: we can assume if idle==true that
+	 * interrupts are already disabled)
+	 */
```

Comment:
FWIW: I'm about to remove the idle parameter here in a separate patch series.  This logic is going into the kernel and will be replaced with a simpler "shut off the counter" API.  Not a problem here, just be aware you might need to rebase this depending on which lands first.